mitosheet/src/components/elements/AuthenticateToSnowflakeCard.tsx

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Mito
+import React, { useEffect, useState } from 'react';
+import MitoAPI from '../../jupyter/api';
+import { classNames } from '../../utils/classNames';
+import { updateObjectWithPartialObject } from '../../utils/objects';
+import '../../../css/layout/CollapsibleSection.css'
+
+import Col from '../layout/Col';
+import CollapsibleSection from '../layout/CollapsibleSection';
+import Row from '../layout/Row';
+import { SnowflakeCredentials } from '../taskpanes/SnowflakeImport/SnowflakeImportTaskpane';
+import Input from './Input';
+import LoadingCounter from './LoadingCounter';
+import TextButton from './TextButton';
+
+export type SnowflakeCredentialsValidityCheckResult = {type: 'success'} | {type: 'error', 'error_message': string}
+
+const getDefaultCredentials = (): SnowflakeCredentials => {
+    return {type: 'username/password', username: '', password: '', account: ''}
+}
+
+/**
+    * A card that handles all work required to authenticate a user to snowflake
+*/ 
+
+const AuthenticateToSnowflakeCard = (props: {
```

Comment:
This card needs some cleanup that is very similar to what I requested on the SnowflakeImportTaskpane last time. With the comments below, the goal is to give you time to think through how to factor it a bit better -- of course, happy to talk about this on a call if it's helpful!

1. The functions are `loadAndSetAndValidateCachedCredentials`, `validateSnowflakeCredentialsParams`, and `_validateSnowflakeCredentials`. Similar feedback to the Snowflake Import taskpane last PR -- I cannot distinguish between these functions, not only what they do, but when I might call them. 
2. The effect sets loading to true, the function `loadAndSetAndValidateCachedCredentials` sets loading to false. But `loadAndSetAndValidateCachedCredentials` is only ever called by the effect!
3. The `validateSnowflakeCredentialsParams` param function does almost nothing - do we need it? If we get rid of it, how does this change how easy it is to name the functions?
4. Related to (3), what happens if we move the setLoading into the `_validateSnowflakeCredentals` function.

Reply:
This is cleaned up due to moving the credentials into the userProfiel